cpp/oneapi/dal/backend/primitives/reduction/test/reduction_dpc.cpp

```diff
@@ -0,0 +1,257 @@
+        const auto gtv = groundtruth_rm_cw();
+        const auto arr = outarr.flatten();
+        check_array(gtv, arr, tol);
+    }
+
+    void check_output_rm_cw(ndarray<float_t, 1, ndorder::c>& outarr, const float_t tol = 1.e-3) {
+        CAPTURE(__func__, width, height, outarr.get_count());
+        const auto gtv = groundtruth_rm_cw();
+        const auto arr = outarr.flatten();
+        check_array(gtv, arr, tol);
+    }
+
+    void check_output_cm_cw(ndarray<float_t, 1, ndorder::c>& outarr, const float_t tol = 1.e-3) {
+        CAPTURE(__func__, width, height);
+        const auto gtv = groundtruth_rm_rw();
+        const auto arr = outarr.flatten();
+        check_array(gtv, arr, tol);
+    }
+
+    void test_rm_rw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(height);
+        auto input =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { height });
+
+        auto reduce_event =
+            reduce_rows(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_rw(output_array);
+    }
+
+    void test_rm_cw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(width);
+        auto input =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { width });
+
+        auto reduce_event =
+            reduce_cols(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_cw(output_array);
+    }
+
+    void test_cm_cw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(height);
+        auto input_tr =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
+        auto input = input_tr.t();
+        auto output =
+            ndview<float_t, 1, ndorder::c>::wrap(output_array.get_mutable_data(), { height });
+
+        auto reduce_event =
+            reduce_cols(get_queue(), input, output, binary_t{}, unary_t{}, { out_event });
+        reduce_event.wait_and_throw();
+
+        check_output_rm_rw(output_array);
+    }
+
+    void test_cm_rw_reduce() {
+        auto input_array = row_accessor<const float_t>{ input_table }.pull(get_queue());
+        auto [output_array, out_event] = output(width);
+        auto input_tr =
+            ndview<float_t, 2, ndorder::c>::wrap(input_array.get_mutable_data(), { height, width });
```

Comment:
`input_array` is array that returned by `row_accessor::pull`. There is no guarantee that it contains mutable data. So many questions here:
1. Why `get_data()` cannot be used here?
2. Does `reduce_rows` assumes mutable data in input?
3. Why tables are needed in this test at all? There is no dependency on table in reduction interface.

Reply:
fixed